src/storage/single_file_block_manager.cpp

```diff
@@ -547,6 +551,28 @@ void SingleFileBlockManager::WriteHeader(DatabaseHeader header) {
 	active_header = 1 - active_header;
 	//! Ensure the header write ends up on disk
 	handle->Sync();
+	// Release the free blocks to the filesystem.
+	TrimFreeBlocks();
+	newly_freed_list.clear();
```

Comment:
```suggestion
```
This is done already as part of TrimFreeBlocks.
Potentially to consider an D_ASSERT that `newly_freed_list is actually empty.